File: src/codex_plugin_scanner/guard/presentation_migration.py
"""One-time migration helpers for obsolete presentation preference values."""

from __future__ import annotations

from typing import Literal

MigratedPresentationMode = Literal["everyday", "technical"]
_LEGACY_PRESENTATION_MODE_MAP: dict[str, MigratedPresentationMode] = {
    "simple": "everyday",
    "advanced": "technical",
    "developer": "technical",
}


def migrate_legacy_presentation_mode(value: object) -> MigratedPresentationMode | None:
    if not isinstance(value, str):
        return None
    return _LEGACY_PRESENTATION_MODE_MAP.get(value)

File: dashboard/src/presentation-migration.ts
import type { GuardPresentationMode } from "./presentation-mode";

const LEGACY_PRESENTATION_MODE_MAP: Readonly<Record<string, GuardPresentationMode>> = {
  simple: "everyday",
  advanced: "technical",
  developer: "technical",
};

export function migrateLegacyPresentationMode(value: unknown): GuardPresentationMode | null {
  return typeof value === "string" ? LEGACY_PRESENTATION_MODE_MAP[value] ?? null : null;
}
